### PR TITLE
Terraform end to end cognito stack

### PR DESCRIPTION
**Pull request by rrrkharse**
**Description of your changes:**
- Creates a VPC, EKS cluster, cognito user pool, Kubeflow platform subdomain (optional or use your own)
- Deploys Kubeflow with the cognito configuration

**Testing:**
- Manual testing
- All pods running, was able to login using the platform domain and run a pipeline

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: deployments/cognito/terraform/cognito-components/main.tf [hunk 1]
```diff
@@ -0,0 +1,270 @@
+provider "aws" {
+  alias = "aws"
+}
+
+provider "aws" {
+  alias = "virginia"
```
Comment (surajkota) [inline deployments/cognito/terraform/cognito-components/main.tf:6]: what about region? alias just a name

File: .github/workflows/terraform-cognito-test.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# todo: test how this will work with the dual input provider changes
```
Comment (surajkota) [inline .github/workflows/terraform-cognito-test.yaml:1]: remove? we will see how it goes anyway 😄

File: deployments/cognito/terraform/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  description = "EKS managed node group status"
+  value       = module.eks_blueprints.managed_node_groups_status
+}
+
+output "configure_kubectl" {
+  description = "Configure kubectl: make sure you're logged in with the correct AWS profile and run the following command to update your kubeconfig"
+  value       = module.eks_blueprints.configure_kubectl
+}
+
+# Region used for Terratest
```
Comment (surajkota) [inline deployments/cognito/terraform/outputs.tf:51]: nit: remove

File: iaac/terraform/aws-infra/cognito/custom_domain.tf [hunk 4]
```diff
@@ -18,13 +26,6 @@ resource "aws_route53_record" "pre_cognito_domain_a_record" {
   }
 }
 
-# Cognito requires a certificate in N.Virginia in order to have a custom domain for a user pool
-# https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-add-custom-domain.html
-provider "aws" {
```
Comment (surajkota) [inline iaac/terraform/aws-infra/cognito/custom_domain.tf:23]: revert?

also lets move this to a separate file called `providers.tf`

**Review by surajkota — CHANGES_REQUESTED** [review]
few questions

File: deployments/cognito/terraform/cognito-components/main.tf (hunk 1 above)
Comment (rrrkharse) [reply deployments/cognito/terraform/cognito-components/main.tf:6]: this is an alias for a provider that is passed into the module, so the provider that is passed in must declare the region (unless that provider itself was an alias)

File: .github/workflows/terraform-cognito-test.yaml (hunk 2 above)
Comment (rrrkharse) [reply .github/workflows/terraform-cognito-test.yaml:1]: haha ok

File: iaac/terraform/aws-infra/cognito/custom_domain.tf (hunk 4 above)
Comment (rrrkharse) [reply iaac/terraform/aws-infra/cognito/custom_domain.tf:23]: doesn't work with this here:
- conflict when passing in the provider
- makes it a legacy module

File: .github/workflows/terraform-cognito-test.yaml (hunk 2 above)
Comment (rrrkharse) [reply .github/workflows/terraform-cognito-test.yaml]: removed

File: deployments/cognito/terraform/outputs.tf (hunk 3 above)
Comment (rrrkharse) [reply deployments/cognito/terraform/outputs.tf]: removed

File: deployments/cognito/terraform/cognito-components/main.tf (hunk 1 above)
Comment (rrrkharse) [reply deployments/cognito/terraform/cognito-components/main.tf:6]: specifically in the root module main.tf there is:
```
provider "aws" {
  region = local.region
}

# Cognito requires a certificate in N.Virginia in order to have a custom domain for a user pool
# https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-add-custom-domain.html
provider "aws" {
  region = "us-east-1"
  alias = "virginia"
}
```

which is wired down to the dependent modules:
```
module "kubeflow_components" {
  source = "./cognito-components"

  ...
  
  providers = {
    aws = aws
    aws.virginia = aws.virginia
  }
}
```

File: iaac/terraform/aws-infra/cognito/custom_domain.tf (hunk 4 above)
Comment (surajkota) [reply iaac/terraform/aws-infra/cognito/custom_domain.tf:23]: ok for now

**Comment by surajkota** [comment]
Thanks, last comment, lets change the action to run make now

**Review by surajkota — APPROVED** [review]

**Pull request merged**